plan/rule_predicate_push_down.go

```diff
@@ -375,3 +392,21 @@ func deriveOtherConditions(p *LogicalJoin, deriveLeft bool, deriveRight bool) (l
 	}
 	return
 }
+
+// conds2TableDual build a LogicalTableDual if cond is constant false or null
+func conds2TableDual(p LogicalPlan, conds []expression.Expression) LogicalPlan {
+	if len(conds) != 1 {
+		return nil
+	}
+	con, ok := conds[0].(*expression.Constant)
+	if !ok {
+		return nil
+	}
+	sc := p.context().GetSessionVars().StmtCtx
+	if isTrue, err := con.Value.ToBool(sc); (err == nil && isTrue == 0) || con.Value.IsNull() {
```

Comment:
Will it be possible that `err != nil && con.Value.IsNull()`

Reply:
It is possible, but the constant is a `NULL` value, the `err` should be ignored.